src/oscar/apps/offer/signals.py

```diff
@@ -0,0 +1,27 @@
+from django.db.models.signals import post_delete
+from django.dispatch import receiver
+
+from oscar.core.loading import get_model
+
+ConditionalOffer = get_model('offer', 'ConditionalOffer')
+Condition = get_model('offer', 'Condition')
+Benefit = get_model('offer', 'Benefit')
+
+
+@receiver(post_delete, sender=ConditionalOffer)
+def delete_related_useless_conditions_and_benefits(instance, **kwargs):
+    offer = instance  # the object is no longer in the database
+
+    condition_id = offer.condition_id
+    condition = Condition.objects.filter(id=condition_id).first()
```

Comment:
The more conventional query here would be `Condition.objects.get(id=conditiong_id)`. Same for the benefit query below.